app/components/autocomplete/channel_mention/channel_mention.js

```diff
@@ -0,0 +1,266 @@
+        },
+        sectionText: {
+            fontSize: 12,
+            color: changeOpacity(theme.centerChannelColor, 0.7),
+            paddingVertical: 7
+        },
+        sectionWrapper: {
+            backgroundColor: theme.centerChannelBg
+        },
+        listView: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg,
+            borderWidth: 1,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.2),
+            borderTopWidth: 0,
+            borderBottomWidth: 0
+        },
+        loading: {
+            alignItems: 'center',
+            justifyContent: 'center',
+            paddingVertical: 20,
+            backgroundColor: theme.centerChannelBg,
+            borderWidth: 1,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.2),
+            borderBottomWidth: 0
+        },
+        row: {
+            padding: 8,
+            flexDirection: 'row',
+            alignItems: 'center',
+            backgroundColor: theme.centerChannelBg,
+            borderTopWidth: 1,
+            borderTopColor: changeOpacity(theme.centerChannelColor, 0.2)
+        },
+        rowDisplayName: {
+            fontSize: 13,
+            color: theme.centerChannelColor
+        },
+        rowName: {
+            color: theme.centerChannelColor,
+            opacity: 0.6
+        }
+    });
+});
+
+export default class ChannelMention extends Component {
+    static propTypes = {
+        currentChannelId: PropTypes.string.isRequired,
+        currentTeamId: PropTypes.string.isRequired,
+        cursorPosition: PropTypes.number.isRequired,
+        autocompleteChannels: PropTypes.object.isRequired,
+        postDraft: PropTypes.string,
+        requestStatus: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        actions: PropTypes.shape({
+            changePostDraft: PropTypes.func.isRequired,
+            autocompleteChannels: PropTypes.func.isRequired
+        })
+    }
+
+    static defaultProps = {
+        postDraft: ''
+    }
+
+    constructor(props) {
+        super(props);
+
+        const ds = new ListView.DataSource({
+            sectionHeaderHasChanged: (s1, s2) => s1 !== s2,
+            rowHasChanged: (r1, r2) => r1 !== r2
+        });
+
+        this.state = {
+            active: false,
+            dataSource: ds.cloneWithRowsAndSections(props.autocompleteChannels)
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const match = nextProps.postDraft.substring(0, nextProps.cursorPosition).match(CHANNEL_MENTION_REGEX);
+
+        // If not match or if user clicked on a channel
+        if (!match || this.state.mentionComplete) {
+            const nextState = {
+                active: false,
+                mentionComplete: false
+            };
+
+            // Handle the case where the user typed a ~ first and then backspaced
+            if (!nextProps.postDraft.length) {
```

Comment:
Does this need to support the case where the user types something like
```
words
~
```
and then backspaces the ~?